Plugins/Public/playercntl_plugin/PimpShip.cpp

```diff
@@ -287,69 +345,220 @@ namespace PimpShip
+
+						if (beginFrom == 0)
+							break;
+					}
+				}
+				else
+				{
+					if (it == firstArg.end() - 1)
+						endAt = mapInfo[iClientID].mapCurrEquip.size();
+					else
+					{
+						endAt = ToUInt(firstArg.substr(index));
+
+						if (endAt < beginFrom)
+							swap(endAt, beginFrom);
+					}
+
+					break;
+				}
+			}
+			else
 			{
-				PrintUserCmdText(iClientID, L"ERR %s", HkErrGetText(err).c_str());
+				PrintUserCmdText(iClientID, L"ERR Invalid syntax");
 				return true;
 			}
-			if (iCash<0 && iCash<set_iCost)
+		}
+
+		if (everyN == 0)
+		{
+			PrintUserCmdText(iClientID, L"ERR Zero advancement");
+			return true;
+		}
+
+		if ((beginFrom == 0) && endAt == 0)
+		{
+			beginFrom = endAt = ToUInt(firstArg);
+			if (beginFrom == 0 || beginFrom > mapInfo[iClientID].mapCurrEquip.size())
 			{
-				PrintUserCmdText(iClientID, L"ERR Insufficient credits");
+				PrintUserCmdText(iClientID, L"ERR hardpoint index is out of bounds");
 				return true;
 			}
-			HkAddCash(wscCharName, 0-set_iCost);
 		}
 
-		// Remove all lights.
-		for (map<uint, EQ_HARDPOINT>::iterator i = mapInfo[iClientID].mapCurrEquip.begin();
-			i != mapInfo[iClientID].mapCurrEquip.end(); ++i)
+		if (beginFrom == 0 || beginFrom > mapInfo[iClientID].mapCurrEquip.size())
 		{
-			pub::Player::RemoveCargo(iClientID, i->second.sID, 1);
+			PrintUserCmdText(iClientID, L"ERR Beginning is out of bounds");
+			PrintUserCmdText(iClientID, L"You may want to use following syntax to select all hardpoints from beginning to %u:", endAt);
+			if (everyN == 1)
+				PrintUserCmdText(iClientID, L"/setitem -%u %u", endAt, iSelectedItemID);
+			else
+				PrintUserCmdText(iClientID, L"/setitem *%u*%u %u", everyN, endAt, iSelectedItemID);
+
+			return true;
 		}
 
-		// Re-add all lights so that the order is kept the same
-		for (map<uint, EQ_HARDPOINT>::iterator i = mapInfo[iClientID].mapCurrEquip.begin();
-			i != mapInfo[iClientID].mapCurrEquip.end(); ++i)
+		if (endAt > mapInfo[iClientID].mapCurrEquip.size())
 		{
-			HkAddEquip(wscCharName, i->second.iArchID, wstos(i->second.wscHardPoint));
+			PrintUserCmdText(iClientID, L"ERR Ending is out of bounds");
+			PrintUserCmdText(iClientID, L"You may want to use following syntax to select all hardpoints from %u to end:", beginFrom);
+			if (everyN == 1)
+				PrintUserCmdText(iClientID, L"/setitem %u- %u", beginFrom, iSelectedItemID);
+			else
+				PrintUserCmdText(iClientID, L"/setitem %u*%u* %u", beginFrom, everyN, iSelectedItemID);
+			return true;
 		}
 
-		PrintUserCmdText(iClientID, L"OK Ship pimp complete. Please wait 10 seconds and reconnect.");
-		HkDelayedKick(iClientID, 5);
-		return true;
+		int totalCost = 0;
+		map<uint, EQ_HARDPOINT>& info = mapInfo[iClientID].mapCurrEquip;
+		uint newItem = mapAvailableItems[iSelectedItemID].iArchID;
+		for (uint i = beginFrom; i < endAt + 1; i += everyN)
+		{
+			if (info[i].iArchID != newItem)
+				totalCost += set_iCost;
+		}
+
+		if (totalCost == 0)
```

Comment:
This will break if the cost is configured as 0.

Reply:
Going to fix.

Reply:
Fixed.